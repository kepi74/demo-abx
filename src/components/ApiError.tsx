import i18next from 'i18next';
import *Â as React from 'react';
import { Alert } from 'reactstrap';

interface IApiErrorProps {
  t: i18next.TFunction;
  error: boolean;
}

const ApiError: React.FC<IApiErrorProps> = ({ error = false, t }) => {
  if (!error) {
    return null;
  }
  return (
    <Alert color="danger">{t('list.errorLoadingData')}</Alert>
  );
};

export default ApiError;
