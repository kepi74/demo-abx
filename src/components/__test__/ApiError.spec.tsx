import i18next from 'i18next';
import * as React from 'react';
import { default as TestRenderer } from 'react-test-renderer';
import ApiError from '../ApiError';

describe('ApiError Component', () => {
  const t = (key: string) =>Â key;

  it('should render error alert if error property is true', () => {
    const tree = TestRenderer
      .create(<ApiError t={t as i18next.TFunction} error={true} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should not render error alert if error property is false', () => {
    const tree = TestRenderer
      .create(<ApiError t={t as i18next.TFunction} error={false} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
