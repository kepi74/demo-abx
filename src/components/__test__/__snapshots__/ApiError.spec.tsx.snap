// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApiError Component should not render error alert if error property is false 1`] = `null`;

exports[`ApiError Component should render error alert if error property is true 1`] = `
<div
  className="alert alert-danger fade"
  role="alert"
>
  list.errorLoadingData
</div>
`;
